Sources/TensorFlow/Core/Runtime.swift

```diff
@@ -1281,6 +1264,32 @@ class _ThreadLocalState {
     }
 }
 
+/// Stack of devices that models nested calls to withDevice/withDefaultDevice. Devices are
+/// represented by their names in TensorFlow notation. See documentation for
+/// `withDevice(named:perform:)` to learn about device names.
+///
+/// All TensorFlow operations will be put on the topmost device on the stack. When the stack is
+/// empty or the topmost device is `nil`, that allows TensorFlow to place operations on any device
+/// that it sees fit.
+@usableFromInline
+class DeviceScopes {
```

Comment:
Why class? is there a reason for multiple threadlocalstate to share the same devicestack?

Reply:
Doesn't have to be. Changed `class` -> `struct` and made the related changes.